src/utils/manifold_features.jl

```diff
@@ -0,0 +1,177 @@
+            push!(features, f)
+        catch
+        end
+    end
+    # d) signatures f(M, p, q)
+    for f in [distance, inverse_retract, log, shortest_geodesic]
+        try
+            f(M, p, p)
+            push!(features, f)
+        catch
+        end
+    end
+    # e) signature f(M, p, X, q)
+    for f in [parallel_transport_to, vector_transport_to]
+        try
+            f(M, p, X, p)
+            push!(features, f)
+        catch
+        end
+    end
+    # f) signature f(M, p, X, Y)
+    for f in [parallel_transport_direction, vector_transport_direction]
+        try
+            f(M, p, X, X)
+            push!(features, f)
+        catch
+        end
+    end
+    # f) signature f(M, p, X, Y, Z)
+    for f in [riemann_tensor]
+        try
+            f(M, p, X, X, X)
+            push!(features, f)
+        catch
+        end
+    end
+    return features
+end
+
+"""
+    ManifoldFeatures
+
+Collect a set of features available on a manifold.
+
+# Fields
+
+* `functions`
+* `retractions`
+* `inverseretractions`
+* `vector_transports`
+* `tolerances` – based on functions or symbols provide a tolerance
+"""
+struct ManifoldFeatures{F<:Function}
+    functions::Vector{F}
+    retractions::Vector{AbstractRetractionMethod}
+    inverse_retractions::Vector{AbstractInverseRetractionMethod}
+    vector_transports::Vector{AbstractVectorTransportMethod}
+end
+function ManifoldFeatures(;
+    functions::Vector{F}=Functions[],
+    retractions=Vector{AbstractRetractionMethod}[],
+    inverse_retractions=Vector{AbstractInverseRetractionMethod}[],
+    vector_transports=Vector{AbstractVectorTransportMethod}[],
+) where {F<:Function}
+    return ManifoldFeatures{F}(
+        functions,
+        retractions,
+        inverse_retractions,
+        vector_transports,
+    )
+end
+function ManifoldFeatures(
+    M::AbstractManifold;
+    functions::Vector{F}=find_manifold_functions(M),
+    retractions=Vector{AbstractRetractionMethod}[],
+    inverse_retractions=Vector{AbstractInverseRetractionMethod}[],
+    vector_transports=Vector{AbstractVectorTransportMethod}[],
+) where {F<:Function}
+    return ManifoldFeatures{F}(
+        functions,
+        retractions,
+        inverse_retractions,
+        vector_transports,
+    )
+end
+
+@doc """
+    ManifoldExpectations
+
+For the features from [`ManifoldFeatures`](@ref) this struct provides expected
+values and tolerances
+
+* `values`     a dictionary `:Symbol->value` for expected values, e.g. `:dimension`
+* `strings`    a dictionary `:Symbol->String` for expected strings, e.g. `:repr` for `repr(M)`
+* `tolerances` a dictionary `:Symbol->Float64` for tolerances in checks of functions.
+"""
+struct ManifoldExpectations
+    values::Dict{Symbol,Int}
+    strings::Dict{Symbol,String}
+    tolerances::Dict{Symbol,Float64}
+end
+function ManifoldExpectations(;
+    values=Dict{Symbol,Int}(),
+    strings=Dict{Symbol,String}(),
+    tolerances=Dict{Symbol,Int}(:default => 1e-14),
```

Comment:
```suggestion
    tolerances=Dict(:default => 1e-14),
```
`Int` is not the best type to specify a tolerance, and if you construct a non-empty `Dict` then types are automatically inferred.

Reply:
ah that was meant to be float.